javalib/src/main/scala/java/net/Inet6Address.scala

```diff
@@ -92,7 +150,16 @@ object Inet6Address {
 
     val ia6ByteArray = in6Addr.getAddress()
 
-    val buffer = new StringBuilder()
+    /* The magic number 64 is used to construct the StringBuffer with a large
+     * enough size that it should not have pay the cost of expanding.
+     * The largest IPv6 address, proper, is 39 ((sizeof("fe80:") * 7) plus 4).
+     * Memory blocks tend to be allocated in powers of two.
+     * Round up to the next higher power of two which will also cover a
+     * possible interface identifier ("%bridge0").
+     * The math need not be exact, the buffer will grow if we guess wrong.
+     */
+
+    val buffer = new StringBuilder(64)
     var isFirst = true
     for (i <- 0 until ia6ByteArray.length) {
```

Comment:
While you are in here, you might want to change this to a normal `while` loop with a `var`.

Reply:
Thank you for the review (and encouragement).

I thought/think that `for()` with a range was optimized in Scala Native quite a long time ago,
perhaps 4 or more years.  Perhaps both ends of the range need to be constants for that to happen?

I will have to try to figure out how to prove to myself if such a loop is optimized.

Reply:
I did find this page so maybe it is ok. I prefer the `while` although more prone to error. https://github.com/scala/bug/issues/1338
```scala
scala> 0 until 10
res0: scala.collection.immutable.Range = Range 0 until 10
```
Whether the Range construction is completely optimized out only byte code would tell.

Reply:
I went with the magic number constant 16. Passes on my build systems. Let's see if CI
likes it.  An Inet6Address is only created by javalib, never users. If the size of a
binary IPv6 address ever changes, it will be evident early in the development process:
things will break all over and in a lot more places that that loop.

Reply:
Eric,

Hard to measure, but your suggestion may have lead to at least a marginal improvement in
the speed of formatting an IP6Address. Those are used all over java.net unit-tests. Once again:
Thank you.